packages/dart_frog_lint/README.md

```diff
@@ -0,0 +1,230 @@
+- [Enabling/disabling lints.](#enablingdisabling-lints)
+  - [Disable one specific rule](#disable-one-specific-rule)
+  - [Disable all lints by default](#disable-all-lints-by-default)
+- [Running dart\_frog\_lint in the terminal/CI](#running-dart_frog_lint-in-the-terminalci)
+- [All the lints](#all-the-lints)
+  - [dart\_frog\_entrypoint](#dart_frog_entrypoint)
+  - [dart\_frog\_middleware](#dart_frog_middleware)
+  - [dart\_frog\_route](#dart_frog_route)
+
+## Installing dart_frog_lint
+
+Dart_frog_lint is implemented using [custom_lint]. As such, it uses custom_lint's installation logic.  
+Long story short:
+
+- Add both dart_frog_lint and custom_lint to your `pubspec.yaml`:
+  ```yaml
+  dev_dependencies:
+    custom_lint:
+    dart_frog_lint:
+  ```
+- Enable `custom_lint`'s plugin in your `analysis_options.yaml`:
+
+  ```yaml
+  analyzer:
+    plugins:
+      - custom_lint
+  ```
+
+## Enabling/disabling lints.
+
+By default when installing dart_frog_lint, most of the lints will be enabled.
+To change this, you have a few options.
+
+### Disable one specific rule
+
+You may dislike one of the various lint rules offered by dart_frog_lint.
+In that event, you can explicitly disable this lint rule for your project
+by modifying the `analysis_options.yaml`
+
+```yaml
+analyzer:
+  plugins:
+    - custom_lint
+
+custom_lint:
+  rules:
+    # Explicitly disable one lint rule
+    - dart_frog_request: false
+```
+
+Note that you can both enable and disable lint rules at once.
+This can be useful if your `analysis_options.yaml` includes another one:
+
+```yaml
+include: path/to/another/analysis_options.yaml
+analyzer:
+  plugins:
+    - custom_lint
+
+custom_lint:
+  rules:
+    # Enable one rule
+    - dart_frog_middleware
+    # Disable another
+    - dart_frog_request: false
+```
+
+### Disable all lints by default
+
+Instead of having all lints on by default and manually disabling lints of your choice,
+you can switch to the opposite logic:  
+Have lints off by default, and manually enable lints.
+
+This can be done in your `analysis_options.yaml` with the following:
+
+```yaml
+analyzer:
+  plugins:
+    - custom_lint
+
+custom_lint:
+  # Forcibly disable lint rules by default
+  enable_all_lint_rules: false
+  rules:
+    # You can now enable one specific rule in the "rules" list
+    - dart_frog_middleware
+```
+
+## Running dart_frog_lint in the terminal/CI
+
+Custom lint rules created by dart_frog_lint may not show-up in `dart analyze`.
+To fix this, you can run a custom command line: `custom_lint`.
+
+Since your project should already have custom_lint installed
+(cf [installing dart_frog_lint](#installing-dart_frog_lint)), then you should be
+able to run:
+
+```sh
+dart run custom_lint
+```
+
+Alternatively, you can globally install `custom_lint`:
+
+```sh
+# Install custom_lint for all projects
+dart pub global activate custom_lint
+# run custom_lint's command line in a project
+custom_lint
+```
+
+## All the lints
```

Comment:
I think this section should be simplified to avoid the README.md growing and growing. I would prefer the documentation for each lint rule to be documented as the class description.

So instead of having:
```dart
/// {@template dart_frog_lint.request}
/// The definition of `dart_frog_request` lints.
/// {@endtemplate}
class DartFrogMiddleware extends DartLintRule {
```

To have something like:
```dart
/// {@template dart_frog_lint.request}
/// The `dart_frog_middleware` lint checks that `routes/*_middleware.dart` files contain a
/// valid `middleware` function. 
///
/// See also:
///
/// * [Middleware](https://dartfrog.vgv.dev/docs/basics/middleware).
///
/// **Good**:
///```dart
/// routes/my_middleware.dart
/// Handler middleware(Handler handler) {
///  return (context) async {
///    // Implement middleware
///  };
/// }
/// ```
///
/// ...
///
/// {@endtemplate}
```

Then including a `lib/lints/lints.dart` where all lint rules are placed under and export the barrel file in the `lib/dart_frog_lint.dart`. So then the `pub.dev` documentation shows something as:

<details>

<summary> Screenshot of dart doc </summary>

![CleanShot 2023-03-22 at 14 01 53@2x](https://user-images.githubusercontent.com/44524995/226928798-a9b6c78a-3342-4136-9194-1f2607fbcff7.png)

</details>

Finally, creating a short list in this `README.md` of the available lints and a link to their documentation page.